### ISSUE TITLE
mypy does not show an error in depends   (fastapi)

### ISSUE THREAD

**Issue by IzyI** [python/mypy#12611]
mypy does not show an error in depends..
common_parameters returns int but str is expected.
```python 

from fastapi import Depends, FastAPI

app = FastAPI()

def common_parameters(i: int) -> int:
    return i * 10

@app.get("/")
async def root(commons: str = Depends(common_parameters)):
    return {"message": commons}
```

Is it possible to solve this problem.

**Comment by ethanhs** [comment]
Looking at FastAPI's source, it looks like `Depends`'s definition returns `Any`, so you lose the type information. I would recommend wrapping the call to depends in your own function to narrow the return type to `int`. You may also want to report this to the FastAPI bug tracker as an enhancement, they likely can use either PEP 612's ParamSpec or if needed a mypy plugin.
